app/views/categories/index.html.erb

```diff
@@ -10,11 +10,19 @@
                 <p><%= category.description %></p>
             </div>
             <div class="c2">
-                <% @lessons = Lesson.where(category_id: category.id, user_id: current_user.id) %>
-                <% if @lessons.any? && @lesson = @lessons.where.not(result: nil).first %>
-                    <%= link_to "Result", @lesson, class: 'btn btn-default float-right' %>
-                <% else %>
+                <% @lesson = Lesson.where(category_id: category.id, user_id: current_user.id).first %>
+                <% if @lesson.blank? %>
```

Comment:
@CharlotteDS  Refactor this area. Use the `current_user.lessons` association to find the necessary the `lesson` object